### ISSUE TITLE
False positive no-empty-first-line-in-method-block for the first local function

### ISSUE THREAD

**Issue by yvolk** [pinterest/ktlint#3336]
For the code:
```kotlin
fun topLevelFunction(
    strParam: String,
    longParam: Long,
): Long {

    fun localFun1(str1: String): Long? = str1.toLongOrNull()
    fun localFun2(str2: String): Long? = str2.take(1).toLongOrNull()

    return localFun1(strParam) ?: localFun2(strParam) ?: longParam
}
```
ktlint correctly complains about absence of a blank line
before declaration `fun localFun2`  
but at the same time it incorrectly complains about PRESENSE of a blank line
before `fun localFun1`.

In my opinion, 
requirement to have a blank line before fun declaration 
[link](https://ktlint.github.io/ktlint/latest/rules/standard/#blank-line-before-declarations)
should be more important than
requirement NOT to have a blank line at the begining of a method block.
[link](https://ktlint.github.io/ktlint/latest/rules/standard/#no-leading-empty-lines-in-method-blocks)

## Expected Behavior
1. Absense of blank line before `fun localFun2` should be marked with  
  "Expected a blank line for this declaration (standard:blank-line-before-declaration)"
2. Blank line before function declaration `fun localFun1` should be considered correct
   as this is the same as above case with
   "Expected a blank line for this declaration (standard:blank-line-before-declaration)"

## Observed Behavior
1. As expected: Absense of blank line before `fun localFun2` is marked with
   "Expected a blank line for this declaration (standard:blank-line-before-declaration)"
2. Bug: Blank line before `fun localFun1` is marked with  
   "First line in a method block should not be empty standard:no-empty-first-line-in-method-block"

## Steps to Reproduce
See the code example above.

## Your Environment
* Version of ktlint used: Ktlint Plugin for IntelliJ IDEA v.0.30.4, ktlint rules version 1.8.0 
* Relevant parts of the `.editorconfig` settings: not present

**Comment by paul-dingemans** [comment]
Hi Yuri,

Thanks for reporting. The blank line before `localFun1` is enforced by rule `standard:no-empty-first-line-in-method-block`. You can [disable the rule](https://ktlint.github.io/ktlint/latest/rules/standard/#no-empty-first-line-at-start-in-class-body) if you do not agree with this rule.

**Comment by yvolk** [comment]
Hello Paul,
I figured out rules that work here. And I agree with the rule `standard:no-empty-first-line-in-method-block` _in general_.
The problem is that the rule should not be applied in THIS particular case. 
And I tried to explain, what is this case and why the rule should not be applied.
Please see my comment.

My goal is to fix code formatting for Kotlin community, not only for me personally.

**Comment by paul-dingemans** [comment]
> The problem is that the rule should not be applied in THIS particular case. And I tried to explain, what is this case and why the rule should not be applied. Please see my comment.
> 
> My goal is to fix code formatting for Kotlin community, not only for me personally.

I appreciate your intend to make it better for the community. 

The Kotlin Coding Conventions are not explicit about this particular situation as the conventions are only guidelines and not strict rules. In such cases Ktlint sometimes takes an opiniated viewpoint inspired on other situations. In this case the reasoning was something like that Ktlint does not favor to have a blank line at the start of class, or a function. This has been generalized to not favoring a blank line at the start of any block.

**Comment by yvolk** [comment]
Thank you for clarification, @paul-dingemans!
Really, I see that current Ktlint behavior is consistent for `fun` nested in `class`, `interface` and in `object` also. I should have checked this using original Ktlint rules.
It appears that, in my custom setup, blank lines at the beginning of `class`, `interface` and `object` blocks ARE allowed! So it looks logical to allow the same for `fun` blocks also. This means that I really need to align my local setup.
